lib/file_transfer_agent/s3_util.js

```diff
@@ -54,11 +54,19 @@ function S3Util(connectionConfig, s3, filestream) {
   this.createClient = function (stageInfo, useAccelerateEndpoint) {
     const stageCredentials = stageInfo['creds'];
     const securityToken = stageCredentials['AWS_TOKEN'];
+    const useS3RegionalUrl = stageInfo.useRegionalUrl || stageInfo.useS3RegionalUrl;
+    
     // if GS sends us an endpoint, it's likely for FIPS. Use it.
     let endPoint = null;
     if (stageInfo['endPoint']) {
-      endPoint = 'https://' + stageInfo['endPoint'];
+      endPoint = `https://${stageInfo['endPoint']}`;
+    } else {
+      if (stageInfo.region && useS3RegionalUrl) {
+        const domainSuffixForRegionalUrl = (stageInfo.region).toLowerCase().startsWith('cn-') ? 'amazonaws.com.cn' : 'amazonaws.com';
```

Comment:
I think we should change the domain in China no matter what's the value of `useS3RegionalUrl ` or `useRegionalUrl` - is there a test to confirm that?

Reply:
The s3 package in node.js has the region field in the configuration. As a result, we need to change the domain when we need to configure the endpoint only. If there is no endPoint, it will create the endPoint based on the region in the configuration by default.